forms-flow-api/src/formsflow_api/services/factory/keycloak_client_service.py

```diff
@@ -93,14 +94,21 @@ def delete_group(self, group_id: str):
             url_path=f"clients/{client_id}/roles/{group_id}"
         )
 
-    def create_group_role(self, data: Dict):
+    @user_context
+    def create_group_role(self, data: Dict, **kwargs):
         """Create role."""
-        client_id = self.client.get_client_id()
-        response = self.client.create_request(
-            url_path=f"clients/{client_id}/roles", data=data
-        )
-        role_name = response.headers["Location"].split("/")[-1]
-        return {"id": role_name}
+        current_app.logger.debug("Creating tenant group...")
+        user: UserContext = kwargs["user"]
+        tenant_key = user.tenant_key
+        # Append tenantKey to group in the group hierarchy.
+        # If user input is "group/sub-group" append tenantKey to each group
+        groups = [group for group in data["name"].split("/") if group]
```

Comment:
Can we add the tenantKey to only main group, as we are doing this to avoid collision and subGroups are unique to group and not per realm ?

Reply:
Added tenantKey to only the main group